scripts/imgtool/image.py

```diff
@@ -69,8 +72,17 @@ class Image():
     @classmethod
     def load(cls, path, included_header=False, **kwargs):
         """Load an image from a given file"""
-        with open(path, 'rb') as f:
-            payload = f.read()
+        ext = os.path.splitext(path)[1][1:]
+        if ext == "hex":
+            ih = IntelHex(path)
```

Comment:
I guess it relies on `open` throwing an exception: https://github.com/bialix/intelhex/blob/master/intelhex/__init__.py#L199

Reply:
Yes, `open` will throw a `FileNotFoundError` if there's no file there. Regarding invalid contents, it will raise an exception itself: https://github.com/bialix/intelhex/blob/master/intelhex/__init__.py#L118

Reply:
And for file format errors:

https://github.com/bialix/intelhex/blob/master/intelhex/__init__.py#L1246

Maybe catching `ValueError` and `IntelHexError` would be enough handling but not sure what's the best way to do it. (I personally would catch those and print the exception `.msg`).

@d3zd3z Suggestions?

Reply:
I don't think we catch exceptions consistently in the rest of the tool?

Reply:
We could catch exceptions at the top level, and try to print something friendly. Honestly, I hate it when programs do that, though, it makes them difficult to debug. I'd much rather just see the exception, even with the stack trace, for something like a corrupt file.

Now "normal" things like a file not found, I think we could catch and print a nice message over.